src/util.js

```diff
@@ -130,6 +130,82 @@ export function mapValues(object, mapper) {
   )));
 }
 
+/**
+ * Formats date and the time
+ *
+ * @param {string} text The text to be formatted
+ * @returns {string} Returns formatted text
+ */
+export function dateTimeFormatting(text) {
+  const locale = 'en-us';
+  const formatOptions = {year: 'numeric', day: 'numeric'};
+
+  const dateRegex = /{{DATE\(\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}:\d{2}Z?([-+]{1}\d{2}:\d{2})?,\s*(SHORT|LONG|COMPACT)\)}}/g;
+  const timeRegex = /{{TIME\(\d{4}-\d{2}-\d{2}T?\d{2}:\d{2}:\d{2}Z?([-+]{1}\d{2}:\d{2})?\)}}/g;
+  const dayRegex = /,(\s{1}.*\s\d{1,2}),/g;
+
+  const dateFormats = {
+    SHORT: 'short',
+    LONG: 'long',
+    COMPACT: 'narrow',
+  };
+
+  function replaceDate(dateMatch, _, type) {
+    const format = dateFormats[type] || dateFormats.COMPACT;
+    const options = {
+      ...formatOptions,
+      weekday: format === dateFormats.COMPACT ? undefined : format,
+      month: format === dateFormats.COMPACT ? 'numeric' : format,
+    };
+    const cleanerRegex = /\(.*,/g;
+    const dateString = dateMatch.match(cleanerRegex)[0].slice(1, -1);
+    const date = new Date(dateString).toLocaleDateString(locale, options);
+
+    return date;
+  }
+
+  function replaceTime(timeMatch) {
+    const cleanerRegex = /\(.*\)/g;
+    const timeString = timeMatch.match(cleanerRegex)[0].slice(1, -1);
+    const time = new Date(timeString).toLocaleTimeString(locale, {hour: '2-digit', minute: '2-digit'});
+
+    return time;
+  }
+
+  function replaceDay(_, match) {
+    let suffix = 'th';
+    const day = match.split(' ')[2];
+    const firstMatches = ['1', '21', '31'];
+    const secondMatches = ['2', '22'];
+    const thirdMatches = ['3', '23'];
+
+    if (firstMatches.indexOf(day) > -1) {
+      suffix = 'st';
+    }
+
+    if (secondMatches.indexOf(day) > -1) {
+      suffix = 'nd';
+    }
+
+    if (thirdMatches.indexOf(day) > -1) {
+      suffix = 'rd';
+    }
+
+    return `,${match}${suffix},`;
+  }
+
+  if (text) {
+    const newText = text
+      .replaceAll(dateRegex, replaceDate)
+      .replaceAll(timeRegex, replaceTime)
+      .replaceAll(dayRegex, replaceDay);
```

Comment:
this will replace in the whole text, not just in the generated date.

I would advise dropping the requirement for day suffixes and just go with the simpler date that .toLocaleDateString() provides